src/libraries/ChainlinkPairDetails.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {ConvertedQuotePair, QuotePair, SortedConvertedQuotePair} from "splits-utils/LibQuotes.sol";
+import {ChainlinkOracleImpl} from "../chainlink/oracle/ChainlinkOracleImpl.sol";
+import {ChainlinkPath} from "./ChainlinkPath.sol";
+
+/// @title Chainlink PairDetails Library
+/// @author 0xSplits
+/// @notice Setters & getters for quote pairs' chainlink details
+library ChainlinkPairDetails {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using ChainlinkPath for bytes;
+
+    /// -----------------------------------------------------------------------
+    /// errors
+    /// -----------------------------------------------------------------------
+    error InvalidFeed_StaleAfter();
+    error InvalidFeed_Decimals();
+    error InvalidFeed_Mul();
+    error InvalidFeed_Empty();
+
+    /// set pair details
+    function _set(
+        mapping(address => mapping(address => ChainlinkOracleImpl.PairDetail)) storage self,
+        function (address) internal view returns (address) convert,
+        ChainlinkOracleImpl.SetPairDetailParams[] calldata params_
+    ) internal {
+        uint256 length = params_.length;
+        for (uint256 i; i < length;) {
+            _validate(params_[i]);
+            _set(self, convert, params_[i]);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// set pair details
+    function _set(
+        mapping(address => mapping(address => ChainlinkOracleImpl.PairDetail)) storage self,
+        function (address) internal view returns (address) convert,
+        ChainlinkOracleImpl.SetPairDetailParams calldata params_
+    ) internal {
+        ConvertedQuotePair memory cqp = params_.quotePair._convert(convert);
+        self[cqp.cBase][cqp.cQuote] = params_.pairDetail;
+        ChainlinkOracleImpl.PairDetail memory pairDetail = params_.pairDetail;
+        pairDetail.inverted = !pairDetail.inverted;
+        self[cqp.cQuote][cqp.cBase] = pairDetail;
+    }
```

Comment:
believe evm has to store from memory, so probably want to save to memory, then save that to storage, then manipulate in memory, then store that
```
        ConvertedQuotePair memory cqp = params_.quotePair._convert(convert);
        ChainlinkOracleImpl.PairDetail memory pairDetail = params_.pairDetail;
        self[cqp.cBase][cqp.cQuote] = pairDetail;
        pairDetail.inverted = !pairDetail.inverted;
        self[cqp.cQuote][cqp.cBase] = pairDetail;
```

Reply:
https://www.evm.codes/?fork=shanghai